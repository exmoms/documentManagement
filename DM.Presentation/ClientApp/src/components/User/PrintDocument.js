import React, { Component } from "react";
import ReactToPrint from "react-to-print";
import Card from "@material-ui/core/Card";
import CardMedia from "@material-ui/core/CardMedia";
import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogTitle from "@material-ui/core/DialogTitle";
import { withTranslation } from "react-i18next";

class PrintDocument extends Component {
  handleCloseDialog = () => {
    this.props.handleClose();
  };

  t = this.props["t"];

  render() {
    const { scannedDoc } = this.props;
    let lang = localStorage.getItem("i18nextLng");
    return (
      <div>
        <Dialog
          open={this.props.open}
          onClose={this.handleCloseDialog}
          aria-labelledby="alert-dialog-title"
          aria-describedby="alert-dialog-description"
        >
          <div
            style={
              lang === "ar" ? { textAlign: "right" } : { textAlign: "left" }
            }
          >
            <DialogTitle id="alert-dialog-title">
              {this.t("doc_print")}
            </DialogTitle>
          </div>

          <DialogContent>
            <Card ref={(el) => (this.componentRef = el)}>
              <React.Fragment>
                {scannedDoc.length !== 0 && (
                  <div>
                    {scannedDoc.map((imageURI, index) => (
                      <div
                        style={{ pageBreakBefore: "always", margin: 25 }}
                        key={index}
                      >
                        <CardMedia
                          component="img"
                          image={imageURI}
                          id="scanned-doc-img-card"
                        />
                      </div>
                    ))}
                  </div>
                )}
              </React.Fragment>
            </Card>
          </DialogContent>
          <DialogActions>
            <ReactToPrint
              trigger={() => (
                <Button color="primary" variant="contained">
                  {this.t("print")}
                </Button>
              )}
              content={() => this.componentRef}
            />
            <Button
              onClick={this.handleCloseDialog}
              color="primary"
              id="handelCloseDialogButton"
            >
              {this.t("cancel")}
            </Button>
          </DialogActions>
        </Dialog>
      </div>
    );
  }
}

export default withTranslation()(PrintDocument);
