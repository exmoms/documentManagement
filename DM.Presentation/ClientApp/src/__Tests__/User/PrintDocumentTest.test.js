import React from "react";
import ReactDom from "react-dom";
import PrintDocument from "../../components/User/PrintDocument";
import { cleanup, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";
import { createMount } from '@material-ui/core/test-utils';
import Dialog from "@material-ui/core/Dialog";
import CardMedia from "@material-ui/core/CardMedia";
import ReactToPrint from "react-to-print";
afterEach(cleanup);
describe("PrintDocument", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  test("renders without crashing", () => {
      let open = true;
      let scannedDoc=["url 1" , "url 2"]
    const handleClose = jest.fn();
    const div = document.createElement("div");
    ReactDom.render(
      <PrintDocument
      open={open}
      scannedDoc={scannedDoc}
      handleClose={handleClose}
      />,
      div
    );
  });
  test("PrintDocument matches snapshot", () => {
    let open = true;
      let scannedDoc=["url 1" , "url 2"]
    const handleClose = jest.fn();
    const renderedValue = createMount()(
        <PrintDocument
        open={open}
        scannedDoc={scannedDoc}
        handleClose={handleClose}
        />
      );
      expect(renderedValue.html()).toMatchSnapshot();
  });
  
  test("test intialization", () => {
    let open = true;
      let scannedDoc=["url 1" , "url 2"]
    const handleClose = jest.fn();
    const wrapper = mount(
        <PrintDocument
        open={open}
        scannedDoc={scannedDoc}
        handleClose={handleClose}
        />
      );
      expect(wrapper.find(Dialog).props().open).toBeTruthy();
      expect(wrapper.find(CardMedia).length).toBe(2);
      expect(wrapper.find(CardMedia).at(0).props().image).toBe("url 1");
      expect(wrapper.find(CardMedia).at(1).props().image).toBe("url 2");
  });

  test("test intialization , handleClose mock  with Dialog", () => {
    let open = true;
      let scannedDoc=["url 1" , "url 2"]
    const handleClose = jest.fn();
    const wrapper = mount(
        <PrintDocument
        open={open}
        scannedDoc={scannedDoc}
        handleClose={handleClose}
        />
      );
      act(() => {
        wrapper
          .find(Dialog)
          .at(0)
          .props()
          .onClose();
      });
      wrapper.update();
      expect(handleClose).toHaveBeenCalledTimes(1);
  });

  test("test intialization , handleClose mock  with the Button", () => {
    let open = true;
      let scannedDoc=["url 1" , "url 2"]
    const handleClose = jest.fn();
    const wrapper = mount(
        <PrintDocument
        open={open}
        scannedDoc={scannedDoc}
        handleClose={handleClose}
        />
      );
      act(() => {
        wrapper
          .find("#handelCloseDialogButton")
          .at(0)
          .props()
          .onClick();
      });
      wrapper.update();
      expect(handleClose).toHaveBeenCalledTimes(1);
  });
});