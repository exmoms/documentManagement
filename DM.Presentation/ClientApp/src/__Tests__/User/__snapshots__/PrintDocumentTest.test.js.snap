// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrintDocument PrintDocument matches snapshot 1`] = `"<div></div>"`;
